lib/ostruct.rb

```diff
@@ -88,12 +89,13 @@ class OpenStruct
   #
   #   data   # => #<OpenStruct country="Australia", capital="Canberra">
   #
-  def initialize(hash=nil)
+  def initialize(hash=nil, is_recursive=false)
     @table = {}
+    @recursive = is_recursive
     if hash
       hash.each_pair do |k, v|
         k = k.to_sym
-        @table[k] = v
+        @table[k] = (recursive && v.is_a?(Hash)) ? OpenStruct.new(v,true) : v
```

Comment:
As I pointed out, best use `respond_to?(:to_hash)` instead of `is_a?(Hash)`

Reply:
Sorry missed that. Updated the code!